### PR TITLE
Prepare for 1.2.0 release

### PR DESCRIPTION
**Pull request by ThomsonTan**
## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,12 +13,24 @@ Increment the:
 * MINOR version when you add functionality in a backwards compatible manner, and
 * PATCH version when you make backwards compatible bug fixes.
 
-## [Unreleased]
```
Comment (reyang) [inline CHANGELOG.md:16]: I guess we want to keep this line? (for future changes)

**Review by reyang — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,12 +13,21 @@ Increment the:
...
+* [API] Allow to use external abseil for bazel targets ([#1172](https://github.com/open-telemetry/opentelemetry-cpp/pull/1172))
+* [EXPORTER] Importing gsl::span if std::span is not available ([#1167](https://github.com/open-telemetry/opentelemetry-cpp/pull/1167))
+* [EXPORTER] Synchronized calls to Exporter::Export & Shutdown ([#1164](https://github.com/open-telemetry/opentelemetry-cpp/pull/1164))
+* [EXPORTER] OTLP http exporter block thread  ([#1163](https://github.com/open-telemetry/opentelemetry-cpp/pull/1163))
+* [TESTS] Jaeger: ThriftSender unit test ([#1162](https://github.com/open-telemetry/opentelemetry-cpp/pull/1162))
+* [EXPORTER] InMemorySpanExporter shutdown fix ([#1161](https://github.com/open-telemetry/opentelemetry-cpp/pull/1161))
+* [EXPORTER] Fix leak in Jaeger exporter ([#1160](https://github.com/open-telemetry/opentelemetry-cpp/pull/1160))
+* [TESTS] ZipkinExporter unit-tests ([#1155](https://github.com/open-telemetry/opentelemetry-cpp/pull/1155))
+* [SDK] `BREAKING CHANGE` Logger: propagating resources through LoggerProvider ([#1154](https://github.com/open-telemetry/opentelemetry-cpp/pull/1154))
```
Comment (reyang) [inline CHANGELOG.md:27]: This is concerning. I think we need to follow the versioning policy https://github.com/open-telemetry/opentelemetry-cpp/blob/main/Versioning.md#api-and-sdk-compatibility.

**Review by reyang — CHANGES_REQUESTED** [review]
The PR suggests that there are breaking changes introduced during a minor version bump. Need to revisit this and make sure we follow https://github.com/open-telemetry/opentelemetry-cpp/blob/main/Versioning.md#api-and-sdk-compatibility.

File: CHANGELOG.md (hunk 2 above)
Comment (lalitb) [reply CHANGELOG.md:27]: The breaking changes are for the logs signal which is an [experimental release](https://github.com/open-telemetry/opentelemetry-cpp/blob/main/Versioning.md#release-policy). So users need to manually enable the feature flag (`WITH_LOGS_PREVIEW` macro) to build and use this functionality, and should be aware that the breaking api/abi changes are possible under the feature flag.

Comment (reyang) [reply CHANGELOG.md:27]: Then it's not breaking change. Is there a way to avoid calling these out as breaking change for production users?

Comment (ThomsonTan) [reply CHANGELOG.md:27]: Yes, it doesn't affect API surface of our stable release. It makes sense to put this notice to a separate section for experimental users without distracting most users.

**Review by lalitb — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -24,11 +24,19 @@ Increment the:
...
 * [EXPORTER] Fix leak in Jaeger exporter ([#1160](https://github.com/open-telemetry/opentelemetry-cpp/pull/1160))
 * [TESTS] ZipkinExporter unit-tests ([#1155](https://github.com/open-telemetry/opentelemetry-cpp/pull/1155))
-* [SDK] `BREAKING CHANGE` Logger: propagating resources through LoggerProvider ([#1154](https://github.com/open-telemetry/opentelemetry-cpp/pull/1154))
-* [SDK] `BREAKING CHANGE` Logger: support for instrumentation library ([#1149](https://github.com/open-telemetry/opentelemetry-cpp/pull/1149))
+* [SDK] Logger: propagating resources through LoggerProvider ([#1154](https://github.com/open-telemetry/opentelemetry-cpp/pull/1154))
+* [SDK] Logger: support for instrumentation library ([#1149](https://github.com/open-telemetry/opentelemetry-cpp/pull/1149))
 * [SDK] Add log level for internal log of sdk ([#1147](https://github.com/open-telemetry/opentelemetry-cpp/pull/1147))
 * [METRICS] Metrics SDK: View API ([#1110](https://github.com/open-telemetry/opentelemetry-cpp/pull/1110))
 
+Note on experimental features:
```
Comment (reyang) [inline CHANGELOG.md:32]:
```suggestion
Notes on experimental features:
```

**Pull request merged**